addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -370,6 +370,41 @@ def set_date_maturity(env):
     )
 
 
+def migrate_bic_numbers(env):
+    """ BIC numbers migrated in supplier bank details """
+    # if no bank exists with a matching BIC, automatically create a new bank, with the new BIC;
+    # if the bank account details do not specify the bank name, use the BIC as bank name.
+    openupgrade.logged_query(
+        env.cr, """
+        WITH banks AS (
+            SELECT ARRAY_AGG(bank_name order by bank_name is null) AS names, bank_bic
```

Comment:
```suggestion
            SELECT ARRAY_AGG(bank_name order by bank_name) AS names, bank_bic
```

Reply:
Your suggestion would give precedence to NULL values instead of proper bank names.

Reply:
This is the SQL ordering + NULL value brainf*ck. By ordering on IS NULL, we get non-null values first because IS NULL will return FALSE for non-null values and FALSE comes before TRUE in the default ascending order.

Reply:
Then it should be:

```suggestion
            SELECT ARRAY_AGG(bank_name ORDER BY bank_name NULLS LAST) AS names, bank_bic
```

Reply:
That would not be wrong, but we don't really care which bank name is selected as long as a non null value is selected when there is one available so the current version is fine really.

Reply:
Well, the problem is in assure that he intended syntax works. The documentation says about "NULLS LAST" but says nothing about "IS NULL" in the ORDER BY clause. I am not saying your syntax doesn't work, just that I prefer using what the documentation says just to be safe.

Reply:
We are sorting here by a boolean expression. It is documented here: https://www.postgresql.org/docs/10/queries-order.html
```
The sort expression(s) can be any expression that would be valid in the query's select list.
```

Reply:
Interesting, good to know. So the difference between `bank_name is null` and `bank_name NULLS LAST` is that the first is nondeterministic, which we don't know which random name would be first, and the second is deterministic, ordering them alphabetically, right?

Reply:
Correct. But given the bank names that people used to assign in the res.partner.bank form, the result is always going to be random ;-)